zeppelin-web/src/app/notebook/notebook.controller.js

```diff
@@ -166,6 +168,16 @@ angular.module('zeppelinWebApp').controller('NotebookCtrl', function($scope, $ro
     }, 10000);
   };
 
+  jQuery(window).on('unload', function(e) {
+    $scope.saveNote();
```

Comment:
Yes, I'm assuming window-unload will be called either on page getting refresh or browser tab getting closed, which will clear up all everything including killSaveTimer.